secant/Wrappers/WrappedSDKSynchronizer.swift

```diff
@@ -160,37 +160,41 @@ class LiveWrappedSDKSynchronizer: WrappedSDKSynchronizer {
         return .none
     }
     
-    func getAllClearedTransactions() -> Effect<[TransactionState], Never> {
+    func getAllClearedTransactions() -> Effect<[WalletEvent], Never> {
         if let clearedTransactions = try? synchronizer?.allClearedTransactions() {
             return Effect(value: clearedTransactions.map {
-                TransactionState.init(confirmedTransaction: $0, sent: ($0.toAddress != nil))
+                let transaction = TransactionState.init(confirmedTransaction: $0, sent: ($0.toAddress != nil))
+                return WalletEvent(id: transaction.id, state: .send(transaction), timestamp: transaction.timestamp)
             })
         }
         
         return .none
     }
     
-    func getAllPendingTransactions() -> Effect<[TransactionState], Never> {
+    func getAllPendingTransactions() -> Effect<[WalletEvent], Never> {
         if let pendingTransactions = try? synchronizer?.allPendingTransactions(),
         let syncedBlockHeight = synchronizer?.latestScannedHeight {
             return Effect(value: pendingTransactions.map {
                 // TODO: - can we initialize it with latestBlockHeight: = nil?
```

Comment:
The answer to this question is "yes, you can, but you shouldn't".

we should rely on either latest height or latest scanned height, whatever is available in that order

Reply:
true, TODO removed